### PR TITLE
[Feature] New Parser and Analyzer support ShowProcStmt

### PR DESCRIPTION
**Pull request by leoyy0316**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/7961

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 19 (63.16%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|3|00.00%|[790, 791, 794]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|[329, 333, 334]|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|4|5|80.00%|[94]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcStmtTest.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+// This file is made available under Elastic License 2.0.
+// This file is based on code available under the Apache license here:
+//   https://github.com/apache/incubator-doris/blob/master/fe/fe-core/src/test/java/org/apache/doris/analysis/DescribeStmtTest.java
```
Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcStmtTest.java:3]: // This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.

Comment (stephen-shelby) [inline fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcStmtTest.java:3]: // This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.

Comment (leoyy0316) [reply fe/fe-core/src/test/java/com/starrocks/analysis/ShowProcStmtTest.java]: done

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 19 (63.16%)




#### file detail

|   |path|covered line|new line|coverage|not covered line detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|3|00.00%|[790, 791, 794]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|[329, 333, 334]|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|4|5|80.00%|[94]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|

**Comment by stephen-shelby** [comment]
cc @HangyuanLiu PTAL

**Review by HangyuanLiu — DISMISSED** [review]

**Comment by leoyy0316** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 12 / 19 (63.16%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|3|00.00%|[815, 816, 819]|
|:large_blue_circle:|com/starrocks/sql/analyzer/ShowStmtAnalyzer.java|5|8|62.50%|[327, 331, 332]|
|:large_blue_circle:|com/starrocks/analysis/ShowProcStmt.java|4|5|80.00%|[94]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|1|1|100.00%|[]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**